### DISCUSSION TITLE
Mermaidが使えない

### DISCUSSION THREAD

**Discussion by glyzinieh** [increments/qiita-discussions#232] [ヘルプ] [answered]
記事の中でmermaidを利用してチャート図をつくったのですが、プレビュー画面にも投稿した記事にも表示されません。
![image](https://user-images.githubusercontent.com/50325323/170509156-35f1d4c2-7012-4118-869d-c6e51d7b6545.png)
![image](https://user-images.githubusercontent.com/50325323/170509609-75f17f26-eaaf-41e0-9ea9-955ed25f0c69.png)

ブラウザのDeveloper ToolでConsoleを見てみると、以下のエラーが表示されていました。
![image](https://user-images.githubusercontent.com/50325323/170509982-7f4e2ec6-2d33-4e2f-92b6-1d0c09119070.png)

**Comment by tanaka-asami415** [comment, marked as answer]
@glyzinieh 
お問い合わせありがとうございます🙏

一度プライベートブラウザでも試してみていただけないでしょうか？
それでも同様の現象が出るようでしたら、以下環境を教えていただけると幸いです🙏

・OS（種類とバージョン ）
・ブラウザ（種類とバージョン ）と拡張機能

**Reply by glyzinieh** [reply]
@tanaka-asami415 
プライベートブラウザでも同じ現象が発生しました。

OSは以下
```
エディション	Windows 11 Home
バージョン	21H2
インストール日	‎2022/‎04/‎29
OS ビルド	22000.675
エクスペリエンス	Windows 機能エクスペリエンス パック 1000.22000.675.0
```
ブラウザは以下の通りです。
![image](https://user-images.githubusercontent.com/50325323/170697805-45d78823-44fd-42fd-ab55-e0ec94d245e0.png)
拡張機能はすべて無効にしてあります。

また、Microsoft Edge・iPhoneのSafariでも同様の現象が確認できました。

**Reply by tanaka-asami415** [reply]
@glyzinieh 
ありがとうございます！

頂いた画像を元に記事を作成してみたところ、我々の環境では起こらなかったため
原文のテキストをそのままいただけないでしょうか？🙏
スペースの全角・半角の違いや、不可視文字などで挙動が違うかもしれないため、試してみようと思います🙏

![image (1)](https://user-images.githubusercontent.com/103916185/170907935-90745ddc-7a64-4fd9-b556-3943ec69731a.png)

**Reply by glyzinieh** [reply]
@tanaka-asami415 
ありがとうございます！

こちらでも試したところ、本文の中に`mermaid`という文字があるとダメなようです。
[このような構成](https://qiita.com/glyzinieh/private/89f1236e17740684f93e)で、前述の現象が発生していたのですが、`mermaid`の部分を消すと正常に機能しました。
````
**（省略）**

### お問い合わせフォーム
説明

### 登録ページ
お問い合わせフォームと同様に、Googleフォームの自作UIを作成。
DBが複数に分かれているため、以下のような構成とした。
```mermaid
graph TD;
    GUI--POST-->サーバー;
    サーバー--POST-->Googleフォーム;
    サーバー--POST-->自宅鯖;
```

### プライバシーポリシー
説明

**（省略）**

### mermaid
説明

**（省略）**
````

**Reply by tanaka-asami415** [reply]
@glyzinieh 
ありがとうございます！

> こちらでも試したところ、本文の中にmermaidという文字があるとダメなようです。

ご指摘の通りこちらが原因となっているようです:eyes:

`Mermaid`の仕様上、Qiitaで`mermaid` という見出しがあると、
内部でMermaidのコードとバッティングしてしまい、上書きされてしまいます💦

申し訳ないのですが、`mermaid`という見出しを別のものにしていただけると幸いです🙏

 [Markdownチートシート](https://qiita.com/Qiita/items/c686397e4a0f4f11683d)へも、`mermaid`という見出しが使えないことを追記しておきます🙇

**Reply by glyzinieh** [reply]
@tanaka-asami415 
了解しました。
対応ありがとうございます！
